docs/ocis/adr/0018-search.md

```diff
@@ -0,0 +1,115 @@
+
+---
+title: "8. oCIS Search Infrastructure"
+date: 2021-06-08T09:00:00+01:00
+geekdocRepo: https://github.com/owncloud/ocis
+geekdocEditPath: edit/master/docs/ocis/adr
+geekdocFilePath: 0008-search.md
+---
+
+* Status: accepted
+* Deciders: @butonic, @micbar, @dragotin
+* Date: 2021-06-08
+
+Technical Story: oCIS Internal Services and APIs for File Search
+
+## Context and Problem Statement
+
+The ability to find files based on certain search terms is a key requirement for a system that provides the ability to store unstructured data large scale. This ADR outlines the concepts to implement search in oCIS.
+
+From the users perspective, the interface to search is just a single entry field where the user enters one or more search terms. The minimum expectation is that the search returns file names and links to files that
+
+- have a file name that contains at least one of the search terms
+- contain at least one of the search terms in the file contents
+- have meta data that is equal or contains one of the search terms
+
+More sophisticated search capabilities are expected and can be implemted, especially based on metadata.
```

Comment:
```suggestion
More sophisticated search capabilities are expected and can be implemented, especially based on metadata.
```

More sophisticated in what way? Search fields like `name:Invoice.pdf max-size:1MB` or something else?